djvm/src/main/java/sandbox/java/util/HashSet.java

```diff
@@ -0,0 +1,87 @@
+package sandbox.java.util;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This is a dummy class that implements just enough of {@link java.util.HashSet}
+ * to allow us to compile {@link sandbox.java.util.AbstractMap}.
+ */
+@SuppressWarnings("WeakerAccess")
+public class HashSet<T> extends sandbox.java.lang.Object implements Set<T> {
+    private static final String UNSUPPORTED = "Dummy class - not implemented";
+    private final java.util.Set<T> set;
+
+    public HashSet(int initialCapacity) {
+        set = new java.util.HashSet<>(initialCapacity);
+    }
+
+    public HashSet() {
+        set = new java.util.HashSet<>();
+    }
+
+    @Override
+    @NotNull
+    public Iterator<T> iterator() {
+        throw new UnsupportedOperationException(UNSUPPORTED);
```

Comment:
I am wondering why this is suppressed and presumably this has to do with undefined order of iteration?
We could simply change the underlying `set` to be `java.util.TreeSet` and make the result deterministic without breaking the contract of this interface, correct?

Reply:
This is a dummy class to bootstrap the compilation of others (see comment at top). Its implementation doesn't really matter.

Reply:
If the implementation is indeed irrelevant then why not throwing `UnsupportedOperationException` for every method? That would clarify that this class is purely a stub.

Reply:
Because `sandbox.java.lang.DJVM.kt` uses a `sandbox.java.lang.LinkedHashMap` to simulate `Enum`, and I test `DJVM.kt` outside the sandbox. So a few basic implementations don't hurt. (You'll notice that most of the dummy classes do indeed throw `UnsupportedOperationException`.)

Reply:
Still slightly confused ;). So tests are still relying on this at runtime? In summary, what are you generating at compile-time vs run-time out of the dummy classes?

Reply:
`sandbox.java.lang.DJVM` needs `sandbox.java.util.LinkedHashMap`, and I have a `DJVMTest` class. So my dummy implementation of `sandbox.java.util.LinkedHashMap` wraps an actual `LinkedHashMap`.

The `sandbox.java.util.concurrent.ConcurrentHashMap` template class wraps a `sandbox.java.util.LinkedHashMap` class so that the DJVM doesn't have any tricky marshalling problems between iterators and spliterators. So the `sandbox.LinkedHashMap` is more complete for that reason too.